0006-cmx2cdp-protocol/README.md

```diff
@@ -1,84 +1,92 @@
 ---
-name: CMX to CDP protocol
-shortname: 6/CMX2CDP
-status: raw
+name: Configuration Data Transport protocol
+shortname: 6/CDT
+status: draft
 editor: Andrew Pasika <apasika@cybervisiontech.com>
 ---
 
-## Introduction
+- [Introduction](#introduction)
+- [Language](#language)
+- [Requirements and constraints](#requirements-and-constraints)
+- [Design](#Design)
+  - [Configuration pull](#configuration-pull)
+    - [Subject structure](#subject-structure)
+    - [Targeted message types](#targeted-message-types)
+  - [Configuration push](#configuration-push)
+    - [Subject structure](#subject-structure)
 
-The CMX2CDP protocol is a [3/Messaging IPC][3/MIPC] protocol extension.
+## Introduction
 
-CMX2CDP is the protocol designed to communicate endpoint configurations from _configuration data provider_
-implementations to the _configuration management extensions_.
+Configuration Data Transport (CDT) protocol is designed to communicate endpoint configuration data between Kaa services.
 
-## Requirements and constraints
-### Problems and possible solutions
+In Kaa architecture, some services are designed to manage endpoint configuration in different ways — store it, process it, publish it, etc.
+One of the core features in such context is the ability to store and retrieve endpoint configurations.
 
-1. _Configuration push._
+For this purpose, the CDT protocol was designed.
+It defines the endpoint configuration data structure so that it's interpreted by all involved services in the same way.
 
-    Solutions:
-    - CDP should broadcast event that new configuration is available.
+In CDT lingo, storing an endpoint configuration to a service is called *configuration push*, while retrieving it from a service is called *configuration pull*.
```

Comment:
Wrong. Configuration push is for pushing updates from the provider service that manages configuration to consumers. Configuration pull is for a consumer to explicitly request the current config state for a certain endpoint from the provider.
Configuration management in the provider is outside of the scope of CDT.